### PR TITLE
(chore) Code quality and tooling improvements

### PR DESCRIPTION
**Pull request by denniskigen**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR includes a number of changes to improve code quality and tooling. These changes include:

- Adding ESLint plugins for testing (`jest-dom`, `testing-library`, `playwright`) and `import` rules for preventing duplicate imports.
- Adding the `react-hooks` ESLint plugin for enforcing rules of hooks and `exhaustive-deps` for React hooks.
- Resolving rules-of-hooks violations flagged by the `react-hooks` ESLint plugin.
- Adapting the `Prettier` configuration to align with similar configurations in O3.
- Adding emojis to steps in the GitHub Actions workflows for improved readability.
- Improving the `react-i18next` test mock to properly handle interpolated values in translation strings.
- Annotating type-only imports with the `type` keyword.

### CODE REVIEW

**Comment by denniskigen** [comment]
FYI, @druchniewicz @pwargulak.

File: packages/esm-reports-app/src/components/overview.component.tsx [hunk 1]
```diff
@@ -24,17 +26,15 @@ import {
...
-import ReportStatus from './report-status.component';
 import { COMPLETED, RAN_REPORT_STATUSES, SAVED } from './report-statuses-constants';
-import ReportOverviewButton from './report-overview-button.component';
+import { DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE, DEFAULT_PAGE_SIZES } from './pagination-constants';
 import { PRIVILEGE_SYSTEM_DEVELOPER } from '../constants';
-import classNames from 'classnames';
+import Overlay from './overlay.component';
+import ReportOverviewButton from './report-overview-button.component';
+import ReportStatus from './report-status.component';
+import RunReportForm from './run-report/run-report-form.component';
```
Comment (NethmiRodrigo) [inline packages/esm-reports-app/src/components/overview.component.tsx:36]: Shouldn't components be before constants? Since components would usually come before types.

File: packages/esm-reports-app/src/components/run-report/cancel-report-modal.component.tsx [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-import { showSnackbar } from '@openmrs/esm-framework';
 import { mutate } from 'swr';
+import { ModalBody, Button, ModalFooter, ModalHeader, InlineLoading } from '@carbon/react';
+import { showSnackbar } from '@openmrs/esm-framework';
 import {
   PROCESSING_REPORT_STATUSES,
   RAN_REPORT_STATUSES,
   SCHEDULED_REPORT_STATUSES,
 } from '../report-statuses-constants';
+import { cancelReportRequest } from '../reports.resource';
```
Comment (NethmiRodrigo) [inline packages/esm-reports-app/src/components/run-report/cancel-report-modal.component.tsx:11]: Same with here, functions before variables seem more natural XD

File: packages/esm-reports-app/src/components/scheduled-overview-cell-content.component.tsx [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 import { useTranslation } from 'react-i18next';
+import { showModal, userHasAccess, useSession } from '@openmrs/esm-framework';
 import { closeOverlay, launchOverlay } from '../hooks/useOverlay';
-import styles from './reports.scss';
 import { PRIVILEGE_SYSTEM_DEVELOPER } from '../constants';
 import EditScheduledReportForm from './edit-scheduled-report/edit-scheduled-report-form.component';
+import NextReportExecution from './next-report-execution.component';
+import ReportOverviewButton from './report-overview-button.component';
+import ReportScheduleDescription from './report-schedule-description.component';
 import ScheduledReportStatus from './scheduled-report-status.component';
```
Comment (NethmiRodrigo) [inline packages/esm-reports-app/src/components/scheduled-overview-cell-content.component.tsx:11]: Ditto

File: packages/esm-reports-app/src/components/scheduled-overview.component.tsx [hunk 4]
```diff
@@ -13,11 +12,12 @@ import {
   TableHeader,
   TableRow,
 } from '@carbon/react';
+import { ExtensionSlot, isDesktop, useLayoutType, usePagination } from '@openmrs/esm-framework';
 import { useScheduledReports } from './reports.resource';
 import { DEFAULT_PAGE_SIZE, DEFAULT_PAGE_SIZES } from './pagination-constants';
 import ScheduledOverviewCellContent from './scheduled-overview-cell-content.component';
 import Overlay from './overlay.component';
```
Comment (NethmiRodrigo) [inline packages/esm-reports-app/src/components/scheduled-overview.component.tsx:19]: Ditto

File: packages/esm-reports-app/src/components/simple-cron-editor/simple-cron-editor.component.tsx [hunk 5]
```diff
@@ -16,12 +15,13 @@ import {
   ST_EVERY_MONTH,
   ST_EVERY_WEEK,
   ST_ONCE,
+  type CronField,
 } from './commons';
+import CronDatePicker from './cron-date-picker.component';
 import CronDayOfMonthSelect from './cron-day-of-month-select.component';
+import CronDayOfWeekSelect from './cron-day-of-week-select.component';
+import CronTimePicker from './cron-time-picker.component';
```
Comment (NethmiRodrigo) [inline packages/esm-reports-app/src/components/simple-cron-editor/simple-cron-editor.component.tsx:23]: Same here

File: packages/esm-reports-app/src/components/scheduled-overview.component.tsx (hunk 4 above)
Comment (denniskigen) [reply packages/esm-reports-app/src/components/scheduled-overview.component.tsx:19]: Cries in "We should have an ESLint rule for this already"

**Comment by denniskigen** [comment]
@NethmiRodrigo I couldn't get tsconfig path aliases working for whatever reason. I specifically wanted to alias the `__mocks__` and `tools` directories. Could you look at it at some point?

**Review by NethmiRodrigo — APPROVED** [review]
LGTM! Thanks, @denniskigen!

**Comment by druchniewicz** [comment]
@denniskigen thanks for quality/code improvements!
can you now merge these two related PRs?:
* https://github.com/openmrs/openmrs-esm-admin-tools/pull/58
* https://github.com/openmrs/openmrs-distro-referenceapplication/pull/891

and one not related to these:
* https://github.com/openmrs/openmrs-esm-patient-communications/pull/5
